### PR TITLE
David.goffredo/fix nginx ingress controller sampling

### PR DESCRIPTION
**Pull request by dgoffredo**
### What does this PR do?
Correct inconsistencies in the "sampling" section of the getting started instructions for tracing the Nginx Ingress Controller for Kubernetes. See https://datadoghq.atlassian.net/browse/APMS-7568

### Motivation
The docs were inaccurate, as a customer learned.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by buraizu** [comment]
Thanks, created DOCS-3891 for tracking

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 1]
```diff
@@ -466,43 +466,36 @@ To set a different service name per Ingress using annotations:
 ```
 The above overrides the default `nginx-ingress-controller.ingress-nginx` service name.
 
-### Sampling
+### Ingress Controller Sampling
 
-To control the volume of traces starting from Nginx that are sent to Datadog, specify a sampling rate by setting the parameter `DD_TRACE_SAMPLING_RULES` to a value between `0.0` (0%) and `1.0` (100%). If no value is specified, 100% of traces starting from Nginx are sent. The Kubernetes Nginx ingress controller uses [v1.2.1][11] of the `dd-opentracing-cpp` library).
```
Comment (dussault-antoine) [inline content/en/tracing/trace_collection/proxy_setup/_index.md:471]: I would keep mentioning that the Ingress Controller uses the v1.2.1 of the opentracing library

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 2]
```diff
@@ -466,43 +466,36 @@ To set a different service name per Ingress using annotations:
...
-  main-snippet: |
-	  env DD_TRACE_SAMPLING_RULES=[];
+To use the [Datadog Agent calculated sampling rates][7] (10 traces per second per Agent by default), specify an empty array of sampling rules:
+```yaml
+env:
+- name: DD_TRACE_SAMPLING_RULES
+  value: '[]'
 ```
 
+If no rules are specified, then sampling defaults to 100%.
```
Comment (dussault-antoine) [inline content/en/tracing/trace_collection/proxy_setup/_index.md:497]: I would call that out earlier in the paragraph

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 3]
```diff
@@ -466,43 +466,36 @@ To set a different service name per Ingress using annotations:
 ```
 The above overrides the default `nginx-ingress-controller.ingress-nginx` service name.
 
-### Sampling
+### Ingress Controller Sampling
 
-To control the volume of traces starting from Nginx that are sent to Datadog, specify a sampling rate by setting the parameter `DD_TRACE_SAMPLING_RULES` to a value between `0.0` (0%) and `1.0` (100%). If no value is specified, 100% of traces starting from Nginx are sent. The Kubernetes Nginx ingress controller uses [v1.2.1][11] of the `dd-opentracing-cpp` library).
+To control the volume of Ingress Controller traces that are sent to Datadog, specify a sampling rule that matches all traces. The `sample_rate` configured in the rule will determine the proportion of traces that are sampled.
```
Comment (dussault-antoine) [inline content/en/tracing/trace_collection/proxy_setup/_index.md:472]:
```suggestion
**Note**: If no rules are specified, then sampling defaults to 100%.

```

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 4]
```diff
@@ -466,43 +466,38 @@ To set a different service name per Ingress using annotations:
...
```
Comment (wantsui) [inline content/en/tracing/trace_collection/proxy_setup/_index.md:471]: Should we call out not to use `datadog-sample-rate`? https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#datadog-sample-rate

**Review by wantsui — APPROVED** [review]
LGTM! I left a small question about `datadog-sample-rate` but think this looks good.

**Review by dussault-antoine — APPROVED** [review]

File: content/en/tracing/trace_collection/proxy_setup/_index.md (hunk 4 above)
Comment (dgoffredo) [reply content/en/tracing/trace_collection/proxy_setup/_index.md:471]: Possibly, but I think that it's better to take the risk and not mention it.

When the most recent release of the Datadog tracer is released in the Ingress Controller (probably within the next few weeks), then `datadog-sample-rate` will work again.

By not mentioning it, these instructions remain valid for both the "broken" and "fixed" releases.

Comment (wantsui) [reply content/en/tracing/trace_collection/proxy_setup/_index.md]: Thanks for clarifying!

**Review by kayayarai — APPROVED** [review]

**Pull request merged**